src/psyclone/tests/alggen_test.py

```diff
@@ -137,9 +141,10 @@ def test_single_function_invoke_qr():
     alg, _ = generate(os.path.join(BASE_PATH,
                                    "1.1.0_single_invoke_xyoz_qr.f90"),
                       api="dynamo0.3")
-    gen = str(alg)
-    assert "USE testkern_qr, ONLY: testkern_qr_type" in gen
-    assert ("CALL invoke_0_testkern_qr_type(f1, f2, m1, a, m2, istp, qr)"
+    gen = str(alg).lower()
+    assert "use testkern_qr, only: testkern_qr_type" not in gen
```

Comment:
Perhaps this test for the removal of `testkern_qr_type` should really just test `use testkern_qr` as at the moment it doesn't pick up whether `use terkern_qr` has been removed. You could add a separate assert or simply change this assert. Similary for all other checks that check a use statement is not in the generated code.

Reply:
Yes, good point.